### PR TITLE
customize metal pool type and name based on platform

### PR DESCRIPTION
**Pull request by sanfilip**

### CODE REVIEW

File: pkg/cli/terraform/metal.go [hunk 1]
```diff
@@ -36,24 +40,59 @@ func NewMetal() *Metal {
 	return &metal
 }
 
-func NewMetalNode() *MetalNode {
+func NewMetalNode(platform paperspace.ClusterPlatformType) *MetalNode {
```
Comment (bbatha) [inline pkg/cli/terraform/metal.go:43]: Shouldn't these be `NewPLATFORMNode` to match the convention?

Comment (sanfilip) [reply pkg/cli/terraform/metal.go:43]: We are conflating AWS, DGX, and Bare Metal node handling in this code already.  The overall module probably should have a different name from the target.  This is a definite shortcoming of not using something like terragrunt.  This metal module acts like both a class and an implementation. :(

If you think it is clearer to change the node constructor name I'll make that change.  Let me know.

Comment (sanfilip) [reply pkg/cli/terraform/metal.go:43]: So maybe the container should be called MetalPlatform as well.  If so, I'd like to do this in a later change.

Comment (sanfilip) [reply pkg/cli/terraform/metal.go:43]: Having reconsidered this, I would propose renaming the whole file to represent that it is not specific to a cloud (aka platform), and move to have specializations for various clouds.  Most of them should have very narrow overrides. ...but in a later refactoring.  Right now this is just a terraform module wrapper, but the module is not a pure specialization either.

Comment (bbatha) [reply pkg/cli/terraform/metal.go:43]: I think changing the name to MetalPlatform seems the most straightforward. The rename of the file seems good as well.

**Review by bbatha — APPROVED** [review]

**Comment by sanfilip** [comment]
fixed history

**Pull request merged**